includes/class-wc-admin-api-init.php

```diff
@@ -454,19 +498,35 @@ public static function orders_lookup_process_batch( $batch_number ) {
 		$order_ids = $order_query->get_orders();
 
 		foreach ( $order_ids as $order_id ) {
-			// TODO: schedule single order update if this fails?
-			WC_Admin_Reports_Orders_Stats_Data_Store::sync_order( $order_id );
-			WC_Admin_Reports_Products_Data_Store::sync_order_products( $order_id );
-			WC_Admin_Reports_Coupons_Data_Store::sync_order_coupons( $order_id );
-			WC_Admin_Reports_Taxes_Data_Store::sync_order_taxes( $order_id );
+			self::orders_lookup_process_order( $order_id );
 		}
 	}
 
 	/**
-	 * Init customers report data store.
+	 * Process a single order to update lookup tables for.
+	 * If an error is encountered in one of the updates, a retry action is scheduled.
+	 *
+	 * @param int $order_id Order ID.
+	 * @return void
 	 */
-	public static function customers_report_data_store_init() {
-		WC_Admin_Reports_Customers_Data_Store::init();
+	public static function orders_lookup_process_order( $order_id ) {
+		$result = array_sum(
+			array(
+				WC_Admin_Reports_Orders_Stats_Data_Store::sync_order( $order_id ),
+				WC_Admin_Reports_Products_Data_Store::sync_order_products( $order_id ),
+				WC_Admin_Reports_Coupons_Data_Store::sync_order_coupons( $order_id ),
+				WC_Admin_Reports_Taxes_Data_Store::sync_order_taxes( $order_id ),
+			)
+		);
+
+		// If all updates were either skipped or successful, we're done.
+		// The update methods return -1 for skip, or a boolean success indicator.
+		if ( 4 === absint( $result ) ) {
+			return;
+		}
+
+		// Otherwise assume an error occurred and reschedule.
+		self::schedule_single_order_process( $order_id );
```

Comment:
Hmm.. are you wanting to see an action fire when an error occurs as a means of debugging? That could be useful.. otherwise, the act of processing a single order will fire all of the actions added to the data stores in #1459 (taxes, stats, products, customers).